model/host/host.go

```diff
@@ -386,21 +386,39 @@ func (h *Host) GetTaskGroupString() string {
 	return fmt.Sprintf("%s_%s_%s_%s", h.RunningTaskGroup, h.RunningTaskBuildVariant, h.RunningTaskProject, h.RunningTaskVersion)
 }
 
-// IdleTime returns how long has this host been idle
+// IdleTime returns how long has this host has been sitting idle.
 func (h *Host) IdleTime() time.Duration {
 	// If the host is currently running a task, it is not idle.
 	if h.RunningTask != "" {
 		return 0
 	}
 
-	// If the host is not running a task then the idle time is the time
-	// elapsed since the last task finished.
-	return h.SinceLastTaskCompletion()
+	// If the host has run a task it's been idle since that task finished running.
+	if h.LastTask != "" {
+		return time.Since(h.LastTaskCompletedTime)
+	}
+
+	// If the host never ran a task it's been idle since the time it was first ready
+	// to run tasks.
```

Comment:
Can we add a comment explaining why idle time is computed differently between the start task agent route (for `IncIdleTime`) and here? Because here, we're intentionally giving some extra time to hosts to come up and ignoring all the idle compute time that it spends waiting for provisioning steps to run. It might not be obvious to a reader that this is an intentional distinction between the calculations (since agent start time always comes after host start time).

Reply:
done